### ISSUE TITLE
Containers cannot be rendered using Axon.Display

### ISSUE THREAD

**Issue by alisinabh** [elixir-nx/axon#340]
Container layers are not rendered properly since they have multiple shapes. See example below.

```elixir
input = Axon.input("input")
l1 = Axon.dense(input, 1)
l2 = Axon.dense(input, 2)
c = Axon.concatenate(l1, l2)

Axon.Display.as_table(c, Nx.template({2, 1}, {:f, 32}))
```
```
** (ArgumentError) invalid dimension in axis 1 found in shape. Each dimension must be a positive integer, but instead got a tensor as dimension: #Nx.Tensor<                                                  
  f32[2][2]                                                                                                                                                                                                   
                                                                                                                                                                                                              
  Nx.Defn.Expr                                                                                                                                                                                                
  parameter a:0                    f32[2][1]                                                                                                                                                                  
  b = max 1.5, 1.0                 f32
  c = divide 3, b                  f32
  d = sqrt c                       f32
  e = negate d                     f32
  f = random_uniform e, d          f32[1][2]
  g = dot a, [1], [], f, [0], []   f32[2][2]
>

This may happen if you are trying to pass a dimension or a shape as an argument to a defn function, for example:

    defn my_defn(dim) do
      Nx.iota({dim})
    end

However, defn treats all arguments as inputs. To address this, you can pass the dimension or the shape as an option instead:

    defn my_defn(opts \ []) do
      opts = keyword(opts, dim: 1) 
      Nx.iota({opts[:dim]})
    end

Invalid shape: {#Nx.Tensor<
   f32[2][1]
   
   Nx.Defn.Expr
   parameter a:0                    f32[2][1]
   b = max 1.0, 1.0                 f32
   c = divide 3, b                  f32
   d = sqrt c                       f32
   e = negate d                     f32
   f = random_uniform e, d          f32[1][1]
   g = dot a, [1], [], f, [0], []   f32[2][1]
 >, #Nx.Tensor<
   f32[2][2]
    
   Nx.Defn.Expr
   parameter a:0                    f32[2][1]
   b = max 1.5, 1.0                 f32
   c = divide 3, b                  f32
   d = sqrt c                       f32
   e = negate d                     f32
   f = random_uniform e, d          f32[1][2]
   g = dot a, [1], [], f, [0], []   f32[2][2]
 >}

    (nx 0.3.0) lib/nx/shape.ex:37: Nx.Shape.validate!/3
    (axon 0.2.0) lib/axon/display.ex:78: Axon.Display.do_axon_to_rows/5
    (axon 0.2.0) lib/axon/display.ex:47: Axon.Display.axon_to_rows/5
    (axon 0.2.0) lib/axon/display.ex:109: anonymous fn/3 in Axon.Display.do_axon_to_rows/5
    (elixir 1.13.4) lib/enum.ex:1715: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    (axon 0.2.0) lib/axon/display.ex:106: Axon.Display.do_axon_to_rows/5
    (axon 0.2.0) lib/axon/display.ex:47: Axon.Display.axon_to_rows/5
    (axon 0.2.0) lib/axon/display.ex:17: Axon.Display.as_table/2
```

**Comment by seanmor5** [comment]
Good catch! I will fix this :)

**Comment by seanmor5** [comment]
Fixed in 179797f8e718957dfebac3bfb113844c75271855
